svm/src/account_loader.rs

```diff
@@ -2227,4 +2371,261 @@ mod tests {
 
         assert_eq!(actual_inspected_accounts, expected_inspected_accounts,);
     }
+
```

Comment:
For posterity, native loader is an edge case because account loading for some reason doesn't reject transactions that invoke the native loader as a program despite the native loader not being a normal "builtin" (therefore doesn't get added to the program cache) as well as not actually existing onchain as an executable account.